pkg/jsonutil/conversion_test.go

```diff
@@ -101,7 +81,7 @@ func verifyJSONToProtoToJSON(t *testing.T, inputJSON string, shouldPreserve bool
 	// Conversion to only JSON escapes characters if specified in options.
 	convertedJSON, err := ProtoToJSON(&proto, options...)
 	assert.NoError(t, err)
-	assert.Equalf(t, shouldPreserve, inputJSON == convertedJSON, "original JSON:\n%s\nconvertedJSON:\n%s", inputJSON, convertedJSON)
+	assert.JSONEq(t, inputJSON, convertedJSON)
```

Comment:
how about checking equality for compat

Reply:
@janisz I do not understand this request. Can you please clarify?

Reply:
Before we compare to be equal, we discussed that compat format should not have additional spaces and order of fields should be same so we may use equal on compat json format.

Reply:
I have added an additional assert.